scripts/pre_commit_linter.py

```diff
@@ -665,119 +682,170 @@ def _get_data(self, filepath, mode):
         return self._CACHE_DATA_DICT[key]
 
 
-def _lint_all_files(
-        js_filepaths, ts_filepaths, py_filepaths, html_filepaths,
-        css_filepaths, verbose_mode_enabled):
-    """This function is used to check if node-eslint dependencies are
-    installed and pass ESLint binary path and lint all the files(JS, Python,
-    HTML, CSS) with their respective third party linters.
+class LintingTaskSpec(python_utils.OBJECT):
```

Comment:
This isn't exactly what we're looking for, since the code here is essentially repackaging the old method. Instead, you want to is to break out each linter (Python, JS, etc) into a separate check like PythonLintChecksManager (but for Pylint). That way, you'll have a list of tasks for each linter operation that you'll pass in.

Reply:
I have made some changes. PTAL! If they are alright.

Reply:
Also, because these third party libraries have their own ways to lint at their fastest (ie. might parallelize on their own), I'd recommend actually having 2 queues. One queue for the third party linters like Pylint and ESLint where the Semaphore number is 1 and another for the custom linting functions where the semaphore number is the default.

Reply:
If you like, you can probably pull out the linters into separate files, and keep this module for just the logic that is common to all linters. (You might not need a manager class in that case.)

Reply:
@kevinlee12 PTAL!

Reply:
@seanlip I think It is better to have linters in this file.

Reply:
@kevinlee12 can you elaborate further?

Reply:
In `lint_all_files`, there are calls to a lot of third party linters. You'll want to pull all those out. Essentially, you'll have to break up this class.

Reply:
@kevinlee12 so you want me to bring `_lint_py_files` under a Class `ThirdPartyPythonLIntChecks` and same for all other third party linters?

Reply:
> I think It is better to have linters in this file.

Could you explain why?

Reply:
> @kevinlee12 so you want me to bring _lint_py_files under a Class ThirdPartyPythonLIntChecks and same for all other third party linters?

They should be in separate classes.

Reply:
@kevinlee12 Maybe because this file contains `BAD_PATTERNS` and `MANDATORY_PATTERNS` definitions which is used by almost all the linters.

Reply:
You can have the bad patterns and mandatory patterns check in this file, and separate all the other filetype-specific lint checks out. Then, what would happen is: the general lint checks in this file run always, and then the specific ones run for the given input filetype. Would that work?

Reply:
@seanlip we also have `EXCLUDED_PATHS` variable and `redirect_stdout` method and many other global variables like `FILES`, `FILE_CACHE`, etc. which are used by every lint check. If we separate all checks then we have to define these in every file.

Reply:
It seems like a good exercise to try and figure out what the dependencies actually are and refactor -- having everything in the same place makes it more likely that you get a "big blob of mud".

Stuff like EXCLUDED_PATHS falls in the "just do this check before passing files to the sub-linters" category. I'm not sure what FILES and FILE_CACHE are used for but it may be a good idea to figure that out and define an interface for what the seam between the main linter and the sub-linters is. Having the sub-linters be in separate files will help with this as it will require you to understand the dependencies properly and find the correct "break" points.